x-pack/plugins/security_solution/public/common/components/guided_onboarding_tour/tour.tsx

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ReactChild } from 'react';
+import React, { createContext, useCallback, useContext, useEffect, useMemo, useState } from 'react';
+
+import useObservable from 'react-use/lib/useObservable';
+import { of } from 'rxjs';
+import { useIsWithinBreakpoints } from '@elastic/eui';
+import { useKibana } from '../../lib/kibana';
+import { securityTourConfig, SecurityStepId } from './tour_config';
+
+export interface TourContextValue {
+  activeStep: number;
+  endTourStep: (stepId: SecurityStepId) => void;
+  incrementStep: (stepId: SecurityStepId, step?: number) => void;
+  isTourShown: (stepId: SecurityStepId) => boolean;
+}
+
+const initialState: TourContextValue = {
+  activeStep: 0,
+  endTourStep: () => {},
+  incrementStep: () => {},
+  isTourShown: () => false,
+};
+
+const TourContext = createContext<TourContextValue>(initialState);
+
+export const TourContextProvider = ({ children }: { children: ReactChild }) => {
+  const { guidedOnboardingApi } = useKibana().services.guidedOnboarding;
+  const isAddDataTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.addData) ?? of(false),
+    false
+  );
+  const isRulesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.rules) ?? of(false),
+    false
+  );
+  const isAlertsCasesTourActive = useObservable(
+    guidedOnboardingApi?.isGuideStepActive$('security', SecurityStepId.alertsCases) ?? of(false),
+    false
+  );
+
+  const tourStatus = useMemo(
+    () => ({
+      [SecurityStepId.addData]: isAddDataTourActive,
+      [SecurityStepId.rules]: isRulesTourActive,
+      [SecurityStepId.alertsCases]: isAlertsCasesTourActive,
+    }),
+    [isAddDataTourActive, isRulesTourActive, isAlertsCasesTourActive]
+  );
+
+  const isSmallScreen = useIsWithinBreakpoints(['xs', 's']);
```

Comment:
I think the check for small screens is not needed anymore. We used it for 8.4 because on small screens, the sidenav is hidden and we couldn't display tour steps pointing to nav items.